src/NzbDrone.Common/EnvironmentInfo/OsInfo.cs

```diff
@@ -25,22 +24,25 @@ public class OsInfo : IOsInfo
 
         static OsInfo()
         {
-            var platform = Environment.OSVersion.Platform;
-
-            switch (platform)
+            if (OperatingSystem.IsWindows())
             {
-                case PlatformID.Win32NT:
-                    {
-                        Os = Os.Windows;
-                        break;
-                    }
-
-                case PlatformID.MacOSX:
-                case PlatformID.Unix:
-                    {
-                        Os = GetPosixFlavour();
-                        break;
-                    }
+                Os = Os.Windows;
+            }
+            else if (OperatingSystem.IsMacOS())
+            {
+                Os = Os.Osx;
+            }
+            else if (OperatingSystem.IsFreeBSD())
```

Comment:
I updated that line to cover both FreeBDS and NetBSD. I think those will catch any other BSD flavour as any other platform would result in a compiler error [here](https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L42).

Reply:
So since I added a check for NetBSD, the build fails with the following error:
```
The platform 'NETBSD' is not a known platform name
```
I believe this is because sonarr currently uses dotnet6 and NetBSD was only added to `OperatingSystem` with dotnet7 (https://github.com/dotnet/runtime/commit/ec894280361230a15f8648969fd73f32e8c45a10).

Looking more into it however, I don't think we need to check for all BSD variants here, as the way `OperatingSystem.IsFreeBSD()` is implemented relies on compile time preprocessor directives, and sonarr currently only produces a build artifact targetting the freebsd-x64 platform. This means that sonarr users running sonarr on any other BSD variant have to be running the freebsd-x64 build artifact, and `OperatingSystem.IsFreeBSD()` will always return true for those artifacts.